apps/frontend/src/pages/analytics/libs/helpers/get-analytics-rows/get-analytics-rows.helper.ts

```diff
@@ -0,0 +1,43 @@
+import { formatDate } from "~/libs/helpers/helpers.js";
+import { type ActivityLogGetAllItemResponseDto } from "~/modules/activity-logs/activity-logs.js";
+
+import { type AnalyticsRow } from "../../types/types.js";
+
+const getAnalyticsRows = (
```

Comment:
Hey, I’ve tried using Objection.js, but it seems limited for complex grouping and aggregation. The only feasible approach I discovered so far involves using more raw Knex methods for grouping and summing data. But I am not sure about it.
Should I use Knex directly? Or alternatively, maybe to fetch the data and then transform it with a helper function?

Reply:
In contributors repository knex was used directly (there also was an issue with objection limitation), so I think you can use it

Reply:
> The only feasible approach I discovered so far involves using more raw Knex methods for grouping and summing data. But I am not sure about it.

It depends on how complicated it will be. The approach with raw queries is preferable here because it will take fewer resources, than grouping on the backend side, also in case we add pagination or additional filters, this may be hard to handle with js grouping. If this will take a lot of time, then let's just move on with the js grouping approach, but do this on the backend side. But if you can implement this with raw queries, this will be better.

Reply:
![image](https://github.com/user-attachments/assets/c2f48548-7f0e-49cf-b9b5-3c66cb1191ca)
Is this format okay? I made it through raw queries. 
But I don't know how to return it correctly, cause the entity doesn't match this data type now?

Reply:
![image](https://github.com/user-attachments/assets/a89f81aa-10e3-4de8-b337-581bf5fe6ef4)
Does this format work? I've tried several, and things are starting to get a bit confusing... I implemented this using a raw query. But now I’m wondering how to work with entities in this format because it no longer matches.

Reply:
I think the previous format was more expected, let's use grouping on the js side then